### PR TITLE
Webpack Upgrade

### PR DESCRIPTION
**Pull request by asabhaney**
Depends on https://github.com/motherinc/communities/pull/194

TODO:

- [x] Upgrade webpack-dev-server: https://github.com/webpack/webpack-dev-server

### CODE REVIEW

File: config/webpack.dev.config.js [hunk 1]
```diff
@@ -14,27 +14,28 @@ module.exports = {
...
+               'css-loader?modules&importLoaders=1&localIdentName=[path]___[name]__[local]___[hash:base64:5]',
+               'resolve-url-loader',
+               'less-loader'
+            ]
+         }
+      ]
    },
    resolve: {
-      extensions: ['', '.js']
+      extensions: ['.js'],
```
Comment (asabhaney) [inline config/webpack.dev.config.js]: @jaredreich Just a heads up, in the upgrade from webpack v1 to v2, there are no blank extensions specified here. Instead they are specified using `enforceExtensions` (below). This was causing the dev build to break.

File: config/webpack.dev.config.js [hunk 2]
```diff
@@ -14,27 +14,28 @@ module.exports = {
...
+      enforceExtension: false
    },
    plugins: [
-      new webpack.optimize.OccurenceOrderPlugin(),
-      new webpack.HotModuleReplacementPlugin(),
-      new webpack.NoErrorsPlugin()
+      new webpack.HotModuleReplacementPlugin()
```
Comment (asabhaney) [inline config/webpack.dev.config.js]: `OccurenceOrderPlugin` is now on by default. Reference: https://webpack.js.org/guides/migrating/#occurrenceorderplugin-is-now-on-by-default

File: config/webpack.dev.config.js (hunk 1 above)
Comment (jaredreich) [inline config/webpack.dev.config.js]: Thx

File: config/webpack.dev.config.js (hunk 2 above)
Comment (jaredreich) [inline config/webpack.dev.config.js]: Good call, thx

**Pull request merged**